src/robot/ros/RosRobot.cpp

```diff
@@ -38,6 +52,262 @@ RosRobot::RosRobot(
+}
+
+//=============================================================================
+void RosRobot::setRosSwitchControllerServiceClient(
+      const std::shared_ptr<::ros::ServiceClient>& rosSwitchControllerServiceClient)
+{
+  mRosSwitchControllerServiceClient = rosSwitchControllerServiceClient;
+}
+
+//=============================================================================
+void RosRobot::setRosJointModeCommandClient(
+      const std::shared_ptr<aikido::control::ros::RosJointModeCommandClient>& rosJointModeCommandClient)
+{
+  mRosJointModeCommandClient = rosJointModeCommandClient;
+}
+
+//=============================================================================
+bool RosRobot::loadController(const std::string loadControllerName)
+{
+  if (!mRosLoadControllerServiceClient)
+  {
+    throw std::runtime_error(
+        "ROS controller manager service client not instantiated.");
+  }
+
+  controller_manager_msgs::LoadController srv;
+  srv.request.name = loadControllerName;
+
+  return mRosLoadControllerServiceClient->call(srv) && srv.response.ok;
+}
+
+//=============================================================================
+bool RosRobot::startController(const std::string startControllerName)
+{
+  return switchControllers(std::vector<std::string>{startControllerName}, std::vector<std::string>{});
+}
+
+//=============================================================================
+bool RosRobot::stopController(const std::string stopControllerName)
+{
+  cancelAllCommands();
+  return switchControllers(std::vector<std::string>{}, std::vector<std::string>{stopControllerName});
+}
+
+//=============================================================================
+bool RosRobot::switchControllers(
+    const std::vector<std::string> startControllersNames, const std::vector<std::string> stopControllersNames)
+{
+  if (!mRosSwitchControllerServiceClient)
+  {
+    throw std::runtime_error(
+        "ROS controller manager service client not instantiated.");
+  }
+
+  controller_manager_msgs::SwitchController srv;
+  // First try stopping the started controllers
+  // Avoids us falsely detecting a failure if already started
+  srv.request.stop_controllers = startControllersNames;
+  srv.request.strictness
+      = controller_manager_msgs::SwitchControllerRequest::BEST_EFFORT;
+  mRosSwitchControllerServiceClient->call(srv); // Don't care about response code
+
+  // Actual command
+  srv.request.start_controllers = startControllersNames;  
+  srv.request.stop_controllers = stopControllersNames;
+  srv.request.strictness
+      = controller_manager_msgs::SwitchControllerRequest::STRICT;
+
+  return mRosSwitchControllerServiceClient->call(srv) && srv.response.ok;
+}
+
+//=============================================================================
+bool RosRobot::switchControllerMode(
+    const hardware_interface::JointCommandModes jointMode)
+{
+  if (!mRosJointModeCommandClient)
+  {
+    throw std::runtime_error(
```

Comment:
Same comment as elsewhere. No need for a runtime error. Print warning and return False